featuretools/primitives/standard/cum_transform_feature.py

```diff
@@ -31,67 +29,51 @@ class CumCount(TransformPrimitive):
 
     def get_function(self):
         def cum_count(values):
-            return values.groupby(values).cumcount() + 1
+            return pd.Series(range(1, len(values) + 1), index=values.index)
```

Comment:
another option that is easier to read would be. not sure which is more performant though

```
values.iloc[:] = range(1, len(values) + 1)
```